src/components/page/FoodsPage.tsx

```diff
@@ -1,3 +1,156 @@
-export const FoodsPage = () => {
-  return <>FoodsPage</>;
+'use client';
+
+import { ContainerIcon, HomeIcon_Off, MenuMeatballIcon } from '@/assets/images/icons';
+import { Icon } from '@/components/ui';
+import { foodCategories } from '@/const/foodCategory';
+import { FilterBadge } from '@/features/foods/components';
+import { FilterButton } from '@/features/foods/components/FilterButton';
+import { FoodList } from '@/features/foods/components/FoodList';
+import { FoodSort } from '@/features/foods/components/FoodSort';
+import { SearchBar } from '@/features/foods/components/SearchBar';
+import { IFoodView } from '@/features/foods/types/utils';
+import {
+  createContainerIdNameMap,
+  createGroupIdNameMap,
+  groupContainersByGroupId,
+} from '@/features/foods/utils/containerMapping';
+import { IContainer, IFood } from '@/types/definition';
+
+import { Route } from 'next';
+import { usePathname, useRouter, useSearchParams } from 'next/navigation';
+import React, { useEffect, useState } from 'react';
+
+export const FoodsPage = ({ containers }: { containers: IContainer[] }) => {
+  const pathname = usePathname();
```

Comment:
### (Optional)
Great job on implementing all the filtering logic!

Creating one or two custom hooks to separate the logic out of this file would make it even better tho, 
are you gonna do it in different scope?

Reply:
Moved to #121 [](url)